### PR TITLE
Make model saving parametric in test_fn

### PR DESCRIPTION
**Pull request by gokceneraslan**
kopt doesn't support model saving for models with custom activation
functions. So in these cases it's not plausible for users to test model
saving with test_fn. Let's make it optional by adding a save_model
argument similar to the one in CompileFN.

Also, users may not wish using test_fn with a test set i.e. a data_fn
that just returns `(x_train, y_train),` tuple. Right now a second set is mandatory
in data_fn. I think we can relax this by just using the training set in test_fn.

### CODE REVIEW

**Comment by Avsecz** [comment]
yup. good idea! thx
